flight/Modules/Stabilization/rate.c

```diff
@@ -0,0 +1,71 @@
+/**
+******************************************************************************
+* @addtogroup TauLabsModules Tau Labs Modules
+* @{
+* @addtogroup StabilizationModule Stabilization Module
+* @{
+*
+* @file       rate.c
+* @author     Tau Labs, http://taulabs.org, Copyright (C) 2015
+* @author     AJ Christensen <aj@junglistheavy.industries>
+* @brief      Rate mode
+*
+* @see        The GNU Public License (GPL) Version 3
+*
+*****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include "openpilot.h"
+
+#include "pid.h"
+#include "stabilization.h"
+#include "stabilizationsettings.h"
+#include "misc_math.h"
+
+//! Private variables
+static float rate_integral[MAX_AXES];
+
+int stabilization_rate(float gyro,
+                       float command,
+                       float *output,
+                       float dT,
+                       bool reinit,
+                       uint32_t axis,
+                       struct pid *pid,
+                       StabilizationSettingsData *settings)
```

Comment:
similarly I suppose consts for `*output`, `*pid`, how might consts be taken of them? Shall I implement the same for `virtaualflybar.c` - currently the same?